### PR TITLE
Fixed report to correspond to needed failing unit test file structure

### PR DESCRIPTION
**Pull request by ShahafBenYakir**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/23791

## Description
Changed failing unit tests file format to support slack notifier

### CODE REVIEW

File: demisto_sdk/commands/lint/lint_manager.py [hunk 1]
```diff
@@ -547,9 +547,8 @@ def report_summary(self, lint_status: dict):
             print(f"{Colors.Fg.red}{wrapper_fail_pack.fill(fail_pack)}{Colors.reset}")
 
     @staticmethod
-    def _create_report(pkgs_status: dict, path: str):
-        if path:
-            json_path = Path(path) / "lint_report.json"
-            json.dump(fp=json_path.open(mode='w'),
-                      obj=pkgs_status,
-                      indent=4)
+    def _create_failed_unit_tests_report(lint_status: dict, path: str):
```
Comment (Itay4) [inline demisto_sdk/commands/lint/lint_manager.py:550]: please add docstring

File: demisto_sdk/commands/lint/tests/linter_manager_test.py [hunk 2]
```diff
@@ -29,3 +31,48 @@ def test_report_failed_image_creation():
                                                           pkgs_status=pkgs_status,
                                                           return_exit_code=EXIT_CODES["image"])
     assert not pkgs_status.called
+
+
+def test_create_failed_unit_tests_report_with_failed_tests():
```
Comment (Itay4) [inline demisto_sdk/commands/lint/tests/linter_manager_test.py:36]: please add test docstring `Given-When-Then`

File: demisto_sdk/commands/lint/tests/linter_manager_test.py [hunk 3]
```diff
@@ -29,3 +31,48 @@ def test_report_failed_image_creation():
...
+    with open(file_path, 'r') as file:
+        content = file.read()
+        fail_list = content.split('\n')
+        assert len(fail_list) == 2
+        assert 'HelloWorld' in fail_list
+        assert 'Infoblox' in fail_list
+    os.remove(file_path)
+
+
+def test_create_failed_unit_tests_report_no_failed_tests():
```
Comment (Itay4) [inline demisto_sdk/commands/lint/tests/linter_manager_test.py:62]: please add test docstring `Given-When-Then`

File: demisto_sdk/commands/lint/lint_manager.py [hunk 4]
```diff
@@ -303,8 +303,8 @@ def run_dev_packages(self, parallel: int, no_flake8: bool, no_bandit: bool, no_m
                              return_exit_code=return_exit_code,
                              skipped_code=skipped_code,
                              pkgs_type=pkgs_type)
-        self._create_report(pkgs_status=pkgs_status,
-                            path=json_report)
+        self._create_failed_unit_tests_report(lint_status=lint_status,
```
Comment (Itay4) [inline demisto_sdk/commands/lint/lint_manager.py:306]: why isnt this in one line?

File: demisto_sdk/commands/lint/lint_manager.py [hunk 5]
```diff
@@ -547,9 +549,26 @@ def report_summary(self, lint_status: dict):
...
+            "fail_packs_pylint": [],
+            "fail_packs_pytest": [],
+            "fail_packs_pwsh_analyze": [],
+            "fail_packs_pwsh_test": [],
+            "fail_packs_image": []
+        }
+        :param path: str
+            The path to save the report.
+        """
+        failed_ut = set().union([second_val for val in lint_status.values() for second_val in val])
```
Comment (GalRabin) [inline demisto_sdk/commands/lint/lint_manager.py:571]:
```suggestion
        failed_packs = set.union(*lint_status.values())
```
** if changed to sets as mentioned above.
the union is a static method so no need for object init, In addition, I think its better to output failed packages for some reason in this report- so you will be notified, there is no reason for failing if something failed we will be notified.
if not so no need for sets at all only `failed_packs = lint_status[""fail_packs_pytest"]`

File: demisto_sdk/commands/lint/lint_manager.py [hunk 6]
```diff
@@ -547,9 +549,26 @@ def report_summary(self, lint_status: dict):
...
     @staticmethod
-    def _create_report(pkgs_status: dict, path: str):
-        if path:
-            json_path = Path(path) / "lint_report.json"
-            json.dump(fp=json_path.open(mode='w'),
-                      obj=pkgs_status,
-                      indent=4)
+    def _create_failed_unit_tests_report(lint_status: dict, path: str):
+        """
+        Creayes and saves a file containing all failed unit tests
```
Comment (Itay4) [inline demisto_sdk/commands/lint/lint_manager.py:554]:
```suggestion
        Creates and saves a file containing all failed unit tests
```

**Pull request merged**